.github/workflows/build-dist.yml

```diff
@@ -0,0 +1,160 @@
+name: build-dist
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    secrets:
+      GH_TOKEN:
+        description: Used by electron-publish\src\gitHubPublisher.ts
+        required: false
+      CSC_LINK:
+        description: Used to codesign binaries
+        required: false
+      CSC_KEY_PASSWORD:
+        description: Used to codesign binaries
+        required: false
+    inputs:
+      os:
+        required: true
+        type: string
+      node:
+        required: false
+        type: string
+        default: ${{ vars.DEFAULT_NODE_VERSION }}
+      MAX_BUILD_ATTEMPTS:
+        required: false
+        type: number
+        default: 5
+      CACHE_DIR:
+        required: false
+        type: string
+        default: dist
+      NPM_NETWORK_TIMEOUT:
+        required: false
+        type: number
+        default: 120000
+
+env:
+  # Disable console advertisements during install phase
+  ADBLOCK: true
+
+jobs:
+  build:
+    name: Build signed artifacts on ${{ inputs.os }}
+    runs-on: ${{ inputs.os }}
+
+    strategy:
+      fail-fast: false
+
+    env:
+      USE_HARD_LINKS: false
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: Calculate build-cache and dist-cache keys
+        # Because the exact same hashFiles invocation produces different
+        # results down the file
+        id: build-cache-keys
+        shell: bash
+        run: |
+          if command -v shasum; then sum() { shasum -a 256 "$@"; } ; else sum() { sha256sum "$@"; }; fi
+          fragment="${{ runner.os }}-${{ inputs.node }}-${{ hashFiles('*.ts', '**/src/**', '**/.npmrc', '**/tsconfig*.json', '**/yarn.lock') }}-$(< package.json sum | command head -c8)"
+          build_fragment="${{ hashFiles('.github/workflows/build.yml') }}"
+          dist_fragment="${build_fragment}-${{ hashFiles('.github/workflows/build-dist.yml') }}"
+          echo "build-key=build-${fragment}-${build_fragment}" | tee -a $GITHUB_OUTPUT
+          echo "dist-key=dist-${fragment}-${dist_fragment}" | tee -a $GITHUB_OUTPUT
+
+      - name: Restore dist cache
+        id: dist_cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ inputs.CACHE_DIR }}
+          key: ${{ steps.build-cache-keys.outputs.dist-key }}
+
+      - name: Restore build cache
+        if: steps.dist_cache.outputs.cache-hit != 'true'
+        id: build_cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ inputs.CACHE_DIR }}
+          key: ${{ steps.build-cache-keys.outputs.build-key }}
+          fail-on-cache-miss: true
```

Comment:
`actions/setup-node@v3` has `cache: yarn`, why do we need manual caching deps with `actions/cache@v3`

Reply:
I'm actually caching build artifacts here.  It reduces the need to rebuild things that have been built, thus speeding up the pipeline.

Reply:
Ok but you have the same if conditions on `yarn compile` and `yarn dist`, so basically this mean that if we have `cache-hit == true` and we changed some code, ci will not build the code, right?